common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -795,13 +796,34 @@ public void registerExecutor(String appId, ExecutorShuffleInfo executorInfo) {
   }
 
   /**
-   * Close the DB during shutdown
+   * Shutdown mergedShuffleCleaner and close the DB during shutdown
    */
   @Override
   public void close() {
+    if (!mergedShuffleCleaner.isShutdown()) {
+      // Use two phases shutdown refer to
+      // https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html
+      try {
+        mergedShuffleCleaner.shutdown();
+        // Wait a while for existing tasks to terminate
+        if (!mergedShuffleCleaner.awaitTermination(10L, TimeUnit.SECONDS)) {
+          mergedShuffleCleaner.shutdownNow();
+          // Wait a while for tasks to respond to being cancelled
+          if (!mergedShuffleCleaner.awaitTermination(10L, TimeUnit.SECONDS)) {
+            logger.warn("mergedShuffleCleaner did not terminate");
+          }
+        }
```

Comment:
We typically do not do both in spark.
Please add a note that we are doing both here because:
a) We want to prevent new tasks and wait for executing tasks to complete gracefully.
b) Once timeout is reached, we want to interrupt running tasks, so that they fail.

This is to prevent updates to rocks db after it is closed.